presto-main/src/main/java/com/facebook/presto/operator/scalar/RowGreaterThanOperator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.facebook.presto.metadata.BoundVariables;
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.metadata.SqlOperator;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+
+import static com.facebook.presto.metadata.Signature.comparableWithVariadicBound;
+import static com.facebook.presto.spi.function.OperatorType.GREATER_THAN;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.util.Reflection.methodHandle;
+
+public final class RowGreaterThanOperator
+        extends SqlOperator
+{
+    public static final RowGreaterThanOperator ROW_GREATER_THAN = new RowGreaterThanOperator();
+    private static final MethodHandle METHOD_HANDLE = methodHandle(RowGreaterThanOperator.class, "greater", Type.class, Block.class, Block.class);
+
+    private RowGreaterThanOperator()
+    {
+        super(GREATER_THAN,
+                ImmutableList.of(comparableWithVariadicBound("T", "row")),
+                ImmutableList.of(),
+                parseTypeSignature(StandardTypes.BOOLEAN),
+                ImmutableList.of(parseTypeSignature("T"), parseTypeSignature("T")));
+    }
+
+    @Override
+    public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type type = boundVariables.getTypeVariable("T");
+        return new ScalarFunctionImplementation(false, ImmutableList.of(false, false), METHOD_HANDLE.bindTo(type), isDeterministic());
+    }
+
+    public static boolean greater(Type rowType, Block leftRow, Block rightRow)
+    {
+        // TODO: Fix this. It feels very inefficient and unnecessary to wrap and unwrap with Block
+        BlockBuilder leftBlockBuilder = rowType.createBlockBuilder(new BlockBuilderStatus(), 1);
+        BlockBuilder rightBlockBuilder = rowType.createBlockBuilder(new BlockBuilderStatus(), 1);
+        rowType.writeObject(leftBlockBuilder, leftRow);
+        rowType.writeObject(rightBlockBuilder, rightRow);
```

Comment:
please at least extract method
```
int compare(Type rowType, , Block leftRow, Block rightRow)
{
       // FIX ME ....
       BlockBuilder leftBlockBuilder = rowType.createBlockBuilder(new BlockBuilderStatus(), 1);
        BlockBuilder rightBlockBuilder = rowType.createBlockBuilder(new BlockBuilderStatus(), 1);
        rowType.writeObject(leftBlockBuilder, leftRow);
        rowType.writeObject(rightBlockBuilder, rightRow);
        return rowType.compareTo(leftBlockBuilder.build(), 0, rightBlockBuilder.build(), 0);
}
```

and reuse it in `greater`, `greaterOrEqual` ... methods.

But it would be better to implement this in a similar fashion as `DecimalInequalityOperators.java`. You could copy that class, drop `opShortShort` method, and implement `opLongLong` as I written above. That copy/paste from decimal would give you between operator as well in around half of the code lines?